src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1019,6 +1019,9 @@ private void checkApiUrlValidity(GitHub github) throws IOException {
             github.checkApiUrlValidity();
         } catch (HttpException e) {
             String message = String.format("It seems %s is unreachable", apiUri == null ? GITHUB_URL : apiUri);
+            AbortException abortException = new AbortException(message);
+            abortException.initCause(e);
+            LOGGER.log(Level.WARNING, "Issue while checking GitHub URL validity", e);
```

Comment:
Why not `throw (abortException)` instead of `throw new AbortException(message);`?

Reply:
AbortException is designed to not have stacktraces and etc. It should clearly say what fails. But seems here @batmat trying to use it in a wrong way.
The right way is to print informative errors to listener (not stacktraces) and then fail with AbortException.

Reply:
and @recena right, this code wouldn't even work.

Reply:
`AbortException` should be thrown (no printing to log) when there is a user-level mistake. It was not appropriate here IMO. Should be something like

```java
} catch (HttpException e) {
    String message = String.format("It seems %s is unreachable", apiUri == null ? GITHUB_URL : apiUri);
    throw new IOException(message, e);
}
```

Reply:
Should be `throw new IOException`.

The critical part is that we should throw an exception. `AbortException` should, in general, be reserved for aborting the job (at least from plugin code). `IOException` and `InterruptedException` should be thrown from the SCMSource implementations during scanning if any IO error or if interrupted respectively.